src/v0/destinations/ga4_v2/transform.ts

```diff
@@ -1,39 +1,54 @@
 import { InstrumentationError, isDefinedAndNotNull } from '@rudderstack/integrations-lib';
-import { ProcessorTransformationRequest, RudderMessage } from '../../../types';
+import { ProcessorTransformationRequest } from '../../../types';
 import { handleCustomMappings } from './customMappingsHandler';
 import { processEvents as ga4Process } from '../ga4/transform';
 import { basicConfigvalidaiton } from '../ga4/utils';
 
 export function process(event: ProcessorTransformationRequest) {
   const { message, destination } = event;
   const { Config } = destination;
+
+  // Create an enhanced config object instead of mutating the original
+  let enhancedConfig = Config;
   if (isDefinedAndNotNull(Config.configData)) {
     if (typeof Config.configData !== 'string') {
       throw new InstrumentationError('Config data is not a string');
     }
     const configDetails = JSON.parse(Config.configData);
-    Config.propertyId = configDetails.PROPERTY;
-    Config.typesOfClient = configDetails.DATA_STREAM.type;
-    if (Config.typesOfClient === 'gtag') {
-      Config.measurementId = configDetails.DATA_STREAM.value;
-    } else if (Config.typesOfClient === 'firebase') {
-      Config.firebaseAppId = configDetails.DATA_STREAM.value;
-    }
-    Config.apiSecret = configDetails.MEASUREMENT_PROTOCOL_SECRET;
-  }
 
-  const eventPayload = message as RudderMessage;
+    // Create a new config object with parsed properties instead of mutating the original
+    enhancedConfig = {
+      ...Config,
+      propertyId: configDetails.PROPERTY,
+      typesOfClient: configDetails.DATA_STREAM.type,
+      apiSecret: configDetails.MEASUREMENT_PROTOCOL_SECRET,
+      ...(configDetails.DATA_STREAM.type === 'gtag' && {
+        measurementId: configDetails.DATA_STREAM.value,
+      }),
+      ...(configDetails.DATA_STREAM.type === 'firebase' && {
+        firebaseAppId: configDetails.DATA_STREAM.value,
+      }),
+    };
+  }
 
-  if (!eventPayload.type) {
+  if (!message.type) {
     throw new InstrumentationError('Message Type is not present. Aborting message.');
   }
 
-  if (eventPayload.type !== 'track') {
-    return ga4Process({ event, destType: 'ga4_v2' });
+  if (message.type !== 'track') {
+    // Create a new event object with the enhanced config for ga4Process
+    const enhancedEvent = {
+      ...event,
+      destination: {
+        ...destination,
+        Config: enhancedConfig,
+      },
+    } as ProcessorTransformationRequest;
+    return ga4Process({ event: enhancedEvent, destType: 'ga4_v2' });
   }
 
-  basicConfigvalidaiton(Config);
+  basicConfigvalidaiton(enhancedConfig);
 
   // custom mappings flow
-  return handleCustomMappings(message, Config);
+  return handleCustomMappings(message, enhancedConfig);
```

Comment:
Ideally we should be able to operate without enchancing the existing config, just providing an extraConfig object to all dependent functions

Reply:
In this case, GA4_v2 is re-using GA4 code where the connection parameters are already part of `Config` object. UX for input of connection parameters is different in GA4 & GA4_v2 - so the additional logic of updating destination config object in GA4_v2.